### PR TITLE
Update to Rust edition 2021

### PR DESCRIPTION
**Pull request by rfuest**
Thank you for helping out with embedded-graphics-simulator development! Please:

- [ ] Check that you've added passing tests and documentation
- [ ] Add an example where applicable
- [ ] Add a `CHANGELOG.md` entry in the **Unreleased** section under the appropriate heading (**Added**, **Fixed**, etc) if your changes affect the **public API**
- [ ] Run `rustfmt` on the project
- [ ] Run `just build` (Linux/macOS only) and make sure it passes. If you use Windows, check that CI passes once you've opened the PR.

## PR description

This PR updates to Rust edition 2021 and the MSRV to 1.57.0.

### CODE REVIEW

File: README.tpl [hunk 1]
```diff
@@ -9,6 +9,11 @@
 
 {{readme}}
 
+## Minimum supported Rust version
+
+The minimum supported Rust version for embedded-graphics is `1.57` or greater.
```
Comment (jamwaffles) [inline README.tpl]: Not sure if this is a copypasta typo?

```suggestion
The minimum supported Rust version for embedded-graphics-simulator is `1.57` or greater.
```

Comment (rfuest) [reply README.tpl]: Oops, it is. Good catch.

**Comment by rfuest** [comment]
The build error reported in #38 should be fixed by this PR. The link check is still reporting an error, but this seems to be a bug in rustdoc.

**Review by jamwaffles — APPROVED** [review]
Looks good. If you're not able to, do you want me to override the build check requirement and merge using my admin powers?

**Comment by rfuest** [comment]
> Looks good. If you're not able to, do you want me to override the build check requirement and merge using my admin powers?

Yes, I'm not allowed to merge it myself.

**Comment by rfuest** [comment]
I've just converted the docs to intra-doc links. Should we disable checklinks and fail the build if rustdoc outputs any warnings? This wouldn't check external links, but this might be a good solution for internal links.

**Comment by jamwaffles** [comment]
Ah excellent, I want to do that in e-g as well (https://github.com/embedded-graphics/embedded-graphics/issues/100). I think we can remove cargo-deadlinks and rely on `cargo doc` to check them all. I'm not too worried about external links as I don't think we check those at the moment anyway. I'd like to see it checking the `#section-heading` links though, if possible.

**Pull request merged**